src/benchmark_tools/runsuites.jl

```diff
@@ -0,0 +1,137 @@
+#
+#  4 - Run configured test
+#      Ex: `runsuite()`
+#      Ex: `runsuite(testcases=cases, testntsteps = ntsteps)`
+#
+#  5 - Plot the results
+#      Ex: `plotsuite()`
+###############################################################################
+
+# functions for running a configured test suite
+"
+runsuite(;testsolvers,testcases,testabstol,testntsteps)
+
+function for easily running test suite scripts
+"
+function runsuite(;testsolvers::Array{Function,1} = [all],
+                                testcases = :all,
+                                testabstols = 10.0.^(-5:-1:-10),
+                                testntsteps = vcat(collect(10.^(1:5)), 500_000),
+                                verbose = false,
+                                progressmeter = true)
+
+    # test all solvers is option is chosen
+    if testsolvers == [all]
+      testsolvers = collect(keys(Solvers.allsolvers))
+    end
+
+    # select the designated test problems
+    testcases = selectcases(testcases)
+
+    # script for testing solvers and cases combinations
+    ## create dictionaries to hold results
+    resODE = Dict{Symbol,Dict}()
+    resODEfixed = Dict{Symbol,Dict}()
+    resSun = Dict{Symbol,Dict}()
+    resDASSL = Dict{Symbol,Any}()
+
+    ## group result dictionaries by solver packages
+    resbysolverpack = Dict{}(Solvers.ODEadaptivesolvers => resODE,
+                            Solvers.ODEfixedsolvers => resODEfixed,
+                            Solvers.sundialssolvers => resSun,
+                            Solvers.DASSLsolvers => resDASSL)
+
+    testreltols = testabstols
+
+    #loop over each solver pack (e.g. ODE, Sundials)
+    for (solverpack,respack) in resbysolverpack
+        ##loop over each test case designated for suite
+        for (n,tc) in testcases
+            res = Dict{Solver,Any}()
+            tstepss = [linspace(tc.tspan[1], tc.tspan[2], n) for n in testntsteps]
+            #loop over each solver designated for suites,
+            #and select the ones in the current solver package
+            for solverfn in testsolvers
+                if solverfn == Solvers.Sundials.idasol || solverfn==Solvers.Sundials.cvode
+                    continue
+                end
+                if haskey(solverpack, solverfn)
+                    solver = solverpack[solverfn]
+                    # run test suite for adaptive or fixed step solver
+
+                    if isapplicable(solver, tc) && isadaptive(solver)
+                        suite = TestSuite(tc, solver, testabstols, testreltols, [NaN])
+                        res[solver] = run_ode_testsuite(suite,verbose = verbose, progressmeter = progressmeter)
+                    elseif isapplicable(solver, tc) && !isadaptive(solver)
+                        if name(tc)==:bruss1d
+                            # 10^5 steps take more than 10 min to run, skip:
+                            suite = TestSuite(tc, solver, tstepss[testntsteps.<50_000])
```

Comment:
Well, this should be an option but not forbidden to run it for more that 10^5 steps.